### PR TITLE
Add vendor: Tektelic withe device: Smart Home Sensor

### PR DESCRIPTION
**Pull request by laurensslats**
#### Summary
Add vendor Tektelic
Add device: Smart Home Sensor

#### Notes for Reviewers
Uncertain if encoder and decoders work
Uncertain if the test payload work

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -1,536 +1,8 @@
 {
   "name": "lorawan-devices",
   "version": "1.0.0",
-  "lockfileVersion": 2,
+  "lockfileVersion": 1,
```
Comment (johanstokking) [inline package-lock.json:4]: Please update your npm version and revert this change

File: vendor/tektelic/T000489x-codec.yaml [hunk 2]
```diff
@@ -0,0 +1,22 @@
+uplinkDecoder:
+  fileName: decoder_home_sensor.js
+  examples:
+    - description: periodic uplink
+      input:
+        fPort: 1
+        bytes: [03, 67, 00, 0A, 04, 68, 28]
```
Comment (johanstokking) [inline vendor/tektelic/T000489x-codec.yaml:7]: Format as hex;
```suggestion
        bytes: [0x03, 0x67, 0x00, 0x0A, 0x04, 0x68, 0x28]
```

Note that `28` is not `0x28`

File: vendor/tektelic/T000489x.yaml [hunk 3]
```diff
@@ -0,0 +1,90 @@
...
+    numeric: 1
+    # Corresponding hardware versions (optional)
+    hardwareVersions:
+      - '1'
+
+    # LoRaWAN Device Profiles per region
+    # Supported regions are EU863-870, US902-928, AU915-928, AS923, CN779-787, EU433, CN470-510, KR920-923, IN865-867, RU864-870
+    profiles:
+      EU863-870:
+        id: SENBNNEU868
```
Comment (johanstokking) [inline vendor/tektelic/T000489x.yaml:22]: Profile IDs must be lowercase, max 36 chars alphanumeric and with dashes only

You can make that restriction documentation part of this pull request

File: vendor/tektelic/T000489x.yaml [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+
+  -  # You can add more firmware versions and use different profiles per version
+
+# Sensors that this device features (optional)
+# Valid values are: accelerometer, altitude, auxiliary, barometer, battery, button, co2, distance, dust, gps, gyroscope,
+# humidity, light, link, magnetometer, moisture, ph, pir, proximity, rssi, snr, sound, temperature, tvoc, velocity,
+# vibration, water, wind direction and wind speed.
+sensors:
+  - temperature
+  - relativeHumidity
```
Comment (johanstokking) [inline vendor/tektelic/T000489x.yaml:38]: This list is exact; so this needs to be `humidity`

File: vendor/tektelic/index.yaml [hunk 5]
```diff
@@ -0,0 +1,4 @@
+# This example contains just one end device: windsensor. It is referenced here in the index.
```
Comment (johanstokking) [inline vendor/tektelic/index.yaml:1]: Wrong comment

File: vendor/tektelic/encoder_home_sensor.js [hunk 6]
```diff
@@ -0,0 +1,700 @@
+function Encoder(data) {
```
Comment (johanstokking) [inline vendor/tektelic/encoder_home_sensor.js:1]: This is the V2 format, which is supported in The Things Stack, but it is not supported in the Device Repository. Please use input and output objects; https://thethingsstack.io/integrations/payload-formatters/javascript/

File: vendor/tektelic/encoder_home_sensor.js [hunk 7]
```diff
@@ -0,0 +1,700 @@
...
+        for (var i = 0; i < len; i++) {
+            result.push(binary_string.charCodeAt(i));
+        }
+        return result;
+    }
+
+    return {
+        'bytes': ret,
+        'port': port
+    };
```
Comment (johanstokking) [inline vendor/tektelic/encoder_home_sensor.js:698]: This is okay, the function signature is wrong

**Review by johanstokking — CHANGES_REQUESTED** [review]

File: vendor/tektelic/T000489x.yaml (hunk 4 above)
Comment (johanstokking) [reply vendor/tektelic/T000489x.yaml:38]: I did this already

File: vendor/tektelic/T000489x.yaml (hunk 3 above)
Comment (johanstokking) [reply vendor/tektelic/T000489x.yaml:22]: I did this already

**Comment by johanstokking** [comment]
> Uncertain if encoder and decoders work
> Uncertain if the test payload work

There was an issue with validation that I fixed. Please rebase against latest `master`, push and you'll see the validation errors with `make validate` (but also here on GitHub if you push the rebased branch)

File: vendor/tektelic/encoder_home_sensor.js (hunk 7 above)
Comment (laurensslats) [reply vendor/tektelic/encoder_home_sensor.js:698]: Didn't manage to update the payload formatters. Will start a discussion about this and update later.

**Comment by laurensslats** [comment]
Processed feedback.

Only didn't manage to update the payload formatters. Need input here from Tektelic or from Ben.

**Comment by johanstokking** [comment]
> @laurensslats Only didn't manage to update the payload formatters. Need input here from Tektelic or from Ben.

See [`4427eaa` (#17)](https://github.com/TheThingsNetwork/lorawan-devices/pull/17/commits/4427eaa7244f3e4925e142ceb35b85fb9cf33b48)

Apart from some minor cleaning, this is the only thing that needs to be done for adapting the functions.

**Comment by johanstokking** [comment]
@laurensslats can you get some more examples from Tektelic?

File: vendor/tektelic/t000489x-profile.yaml [hunk 8]
```diff
@@ -0,0 +1,8 @@
+vendorProfileID: 415
+supportsClassB: false
+supportsClassC: false
+macVersion: 1.0.3
+regionalParametersVersion: RP001-1.0.3-RevA
+supportsJoin: true
+maxEIRP: 16
```
Comment (johanstokking) [inline vendor/tektelic/t000489x-profile.yaml:7]: Note that this is for EU868

File: vendor/tektelic/t000489x.yaml [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+
+    # LoRaWAN Device Profiles per region
+    # Supported regions are EU863-870, US902-928, AU915-928, AS923, CN779-787, EU433, CN470-510, KR920-923, IN865-867, RU864-870
+    profiles:
+      EU863-870:
+        id: t000489x-profile
+        lorawanCertified: false
+        codec: t000489x-codec
+      US902-928:
+        id: t000489x-profile
```
Comment (johanstokking) [inline vendor/tektelic/t000489x.yaml:26]: So this one needs a different profile with 27 dBm

**Review by johanstokking — CHANGES_REQUESTED** [review]
Also add some examples

**Comment by benolayinka** [comment]
@laurensslats let me know if Tektelik doesn't help and I can rewrite your JS for you

**Comment by johanstokking** [comment]
Also let me know when this is ready for review by requesting one

**Comment by johanstokking** [comment]
And thou shalt always set a milestone

File: vendor/tektelic/encoder_home_sensor.js [hunk 10]
```diff
@@ -0,0 +1,700 @@
...
```
Comment (johanstokking) [inline vendor/tektelic/encoder_home_sensor.js:698]:
```suggestion
    return {
        bytes: ret,
        fPort: port
    };
```

**Review by johanstokking — CHANGES_REQUESTED** [review]
- The `package-lock.json` file shouldn't cause a diff. Please update `npm`
- I see both `smart-home-sensor.jpeg` and `.jpg` checked in. Pick one and remove the other
- Add downlink encoder example
- Probably better to rename the `{encoder,decoder}_home_sensor` to `t00048xx-codec.js` (and put both functions in one file for maintainability)

**Comment by htdvisser** [comment]
Waiting for input from Tektelic

File: vendor/tektelic/encoder_home_sensor.js (hunk 10 above)
Comment (laurensslats) [reply vendor/tektelic/encoder_home_sensor.js:698]: Processed suggested change.

> - Add downlink encoder example
Will need input from Tektelic for this

> - Probably better to rename the {encoder,decoder}_home_sensor to t00048xx-codec.js (and put both functions in one file for maintainability)
Not sure how this should work, doesn't seem essential

**Comment by laurensslats** [comment]
Processed all feedback. As of now, it's ready to merge into the device repo. I've requested Tektelic to review, we can update content if Tektelic has feedback after merging.

**Review by johanstokking — APPROVED** [review]
